### ISSUE TITLE
Add support for custom dev properties

### ISSUE THREAD

**Issue by zvonimirfras** [IBM/carbon-ui-builder#153]
</> tab should have a way of adding (and removing) multiple custom properties

It needs an `id` and a `value`.

`id` should be prefilled with `data-` (as html custom properties) but editable
`value` is a string

(feature requested by devs for testing - `data-testid`)

**Comment by codingmickey** [comment]
Hi @zvonimirfras, does this need to be like for each element to change their properties?
I don't get what you are expecting over here

**Comment by zvonimirfras** [comment]
> Hi @zvonimirfras, does this need to be like for each element to change their properties? I don't get what you are expecting over here

Hi Kartik (@codingmickey)!

Contextual code tab has code related properties for the selected component. Example:
<img width="300" alt="image" src="https://user-images.githubusercontent.com/9692126/195356631-4cd8e745-722a-47a0-864a-234a8f956971.png">

Talking to some of the users surfaced the need for adding custom properties.
They'd use this to add extra attributes to the DOM elements for things like test-ids (that they could use to write tests).

To close this issue, this needs

- UI for adding and removing these properties
- they should be stored in the component JSON
- exported as attributes in both of the supported frameworks.

Makes sense?

_If you want to get into more specifics, feel free to ping me on discord (`Zvonimir#9989`)_
_Or just ask here._

**Comment by codingmickey** [comment]
Yes got it!! Thanks a lot, @zvonimirfras for the detailed overview. I will start working on this in my free time. Firstly will try to get comfortable with the code base.

Sure! I will ask my doubts curiously if I come across any 👍🏽

**Comment by zvonimirfras** [comment]
Great, @codingmickey! Since this requires some design, and we follow Carbon Design System guidelines, please share a sketch before you start coding, to avoid needing design do-overs. _(you can reach out to me privately too, I'm not hard to find)_

You can even make it in UI Builder, to go all meta 😛

**Comment by codingmickey** [comment]
hi @zvonimirfras sorry for the inactivity I totally forgot about this issue.. does this still need help?
